lightning/src/chain/keysinterface.rs

```diff
@@ -1074,12 +1080,14 @@ impl KeysManager {
 					}
 					spend_tx.input[input_idx].witness = keys_cache.as_ref().unwrap().0.sign_dynamic_p2wsh_input(&spend_tx, input_idx, &descriptor, &secp_ctx)?;
 				},
-				SpendableOutputDescriptor::StaticOutput { ref output, .. } => {
-					let derivation_idx = if output.script_pubkey == self.destination_script {
+				SpendableOutputDescriptor::StaticOutput { ref output, channel_keys_id,.. } => {
+					let destination_script = channel_keys_id.map_or_else(|| self.destination_script.clone(),|s| self.get_destination_script(s));
+					let derivation_idx = if output.script_pubkey == destination_script || output.script_pubkey == self.destination_script {
 						1
 					} else {
 						2
 					};
+					
 					let secret = {
```

Comment:
Doesn't the derivation here need to change to match the new style? Also the current code only has two possible `derivation_idx` values, but you'll need three - if `channel_keys_id` is set you need to be able to do both new-style derivation and support both of the two old-style derivation values.